pkg/cmd/util/clientcmd/factory.go

```diff
@@ -33,24 +36,73 @@ import (
 	routegen "github.com/openshift/origin/pkg/route/generator"
 )
 
+// defaultClusterConfigURL is a local name that is used to identify when the client config
+// is unspecified.
+const defaultClusterConfigURL = "https://localhost:8443"
+
 // New creates a default Factory for commands that should share identical server
 // connection behavior. Most commands should use this method to get a factory.
 func New(flags *pflag.FlagSet) *Factory {
-	// Override global default to "" so we force the client to ask for user input
 	// TODO refactor this upstream:
 	// DefaultCluster should not be a global
 	// A call to ClientConfig() should always return the best clientCfg possible
 	// even if an error was returned, and let the caller decide what to do
-	kclientcmd.DefaultCluster.Server = ""
+	kclientcmd.DefaultCluster.Server = defaultClusterConfigURL
 
 	// TODO: there should be two client configs, one for OpenShift, and one for Kubernetes
 	clientConfig := DefaultClientConfig(flags)
+	clientConfig = defaultingClientConfig{clientConfig}
 	f := NewFactory(clientConfig)
 	f.BindFlags(flags)
 
 	return f
 }
 
+// defaultingClientConfig detects whether the provided config is the default, and if
+// so returns an error that indicates the user should set up their config.
+type defaultingClientConfig struct {
+	nested kclientcmd.ClientConfig
+}
+
+// RawConfig calls the nested method
+func (c defaultingClientConfig) RawConfig() (kclientcmdapi.Config, error) {
+	return c.nested.RawConfig()
+}
+
+// Namespace calls the nested method
+func (c defaultingClientConfig) Namespace() (string, bool, error) {
+	return c.nested.Namespace()
+}
+
+// ClientConfig returns a complete client config
+func (c defaultingClientConfig) ClientConfig() (*kclient.Config, error) {
+	cfg, err := c.nested.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	if isDefaultConfig(cfg) {
```

Comment:
So if we actually want to solve it this way, why not try to do it upstream?  I can't say that I'm really a fan of this.